### ISSUE TITLE
BertAug on Insert action issue

### ISSUE THREAD

**Issue by neerajvashistha** [makcedward/nlpaug#1]
Hi Edward,

I was exploring your repository and kudos for all the awesome work you have done! But I faced this issue when I ran     `naw.BertAug(action=Action.INSERT),` in a loop over a large set of sentences, 
__
```
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-8-0b61f9f70462> in <module>
      1 # df['questions'].apply(lambda x: aug.augment(x) if len(x.split(' ')) > 4 and len(x.split(' ')) < 20 else x)
      2 for i in range(len(df.index)):
----> 3     print(df['questions'][i],len(df['questions'][i].split(' ')),aug.augment(df['questions'][i]) if len(df['questions'][i].split(' ')) > 4 and len(df['questions'][i].split(' ')) < 20 else df['questions'][i])

~/bot/bot-ml/lib/python3.7/site-packages/nlpaug/flow/sometimes.py in augment(self, text)
     20                     continue
     21 
---> 22                 augmented_text = aug.augment(augmented_text)
     23 
     24             results.append(augmented_text)

~/bot/bot-ml/lib/python3.7/site-packages/nlpaug/base_augmenter.py in augment(self, tokens)
     42     def augment(self, tokens):
     43         if self.action == Action.INSERT:
---> 44             return self.insert(tokens)
     45         elif self.action == Action.SUBSTITUTE:
     46             return self.substitute(tokens)

~/bot/bot-ml/lib/python3.7/site-packages/nlpaug/augmenter/word/bert.py in insert(self, text)
     60         for aug_idx in aug_idexes:
     61             results.insert(aug_idx, nml.Bert.MASK)
---> 62             new_word = self.sample(self.model.predict(results, nml.Bert.MASK, self.aug_n), 1)[0]
     63             results[aug_idx] = new_word
     64 

~/bot/bot-ml/lib/python3.7/site-packages/nlpaug/base_augmenter.py in sample(self, x, num)
     67 
     68     def sample(self, x, num):
---> 69         return random.sample(x, num)
     70 
     71     def generate_aug_cnt(self, size):

~/bot/bot-ml/lib/python3.7/random.py in sample(self, population, k)
    319         n = len(population)
    320         if not 0 <= k <= n:
--> 321             raise ValueError("Sample larger than population or is negative")
    322         result = [None] * k
    323         setsize = 21        # size of a small set minus size of an empty list

ValueError: Sample larger than population or is negative
```

If i understand correctly, sample is getting k = 1, population as the result of self.model.predict(results, nml.Bert.MASK, self.aug_n) and n is length of population , and if the predictions(length of population) are less than 1 then this is failing with above error.

**Comment by makcedward** [comment]
@neerajvashistha 

Thank you for your appreciation.

Your investigation is correct. If input is empty, this error will be thrown from random package. Since no input is provided, BERT prediction will not work even though random selection phase is passed. 

Target to include validation before augmentation. If there is invalidate input (e.g. empty input), empty result will be return. Does it make sense? Or do you have suggestion?

**Comment by neerajvashistha** [comment]
Yes, this is exactly my thoughts were, well you have written all the code on similar lines, this will have an effect on other predictions too. for example 
```
~/bot/bot-ml/lib/python3.7/site-packages/nlpaug/flow/sometimes.py in augment(self, text)
     20                     continue
     21 
---> 22                 augmented_text = aug.augment(augmented_text)
     23 
     24             results.append(augmented_text)

~/bot/bot-ml/lib/python3.7/site-packages/nlpaug/base_augmenter.py in augment(self, tokens)
     44             return self.insert(tokens)
     45         elif self.action == Action.SUBSTITUTE:
---> 46             return self.substitute(tokens)
     47         elif self.action == Action.DELETE:
     48             return self.delete(tokens)

~/bot/bot-ml/lib/python3.7/site-packages/nlpaug/augmenter/word/word_embs_aug.py in substitute(self, text)
     54         word_idxes = [i for i, t in enumerate(tokens)]
     55         word_idxes = self.skip_aug(word_idxes, tokens)
---> 56         aug_idexes = self.sample(word_idxes, aug_cnt)
     57 
     58         for aug_idx in aug_idexes:

~/bot/bot-ml/lib/python3.7/site-packages/nlpaug/base_augmenter.py in sample(self, x, num)
     67 
     68     def sample(self, x, num):
---> 69         return random.sample(x, num)
     70 
     71     def generate_aug_cnt(self, size):

~/bot/bot-ml/lib/python3.7/random.py in sample(self, population, k)
    319         n = len(population)
    320         if not 0 <= k <= n:
--> 321             raise ValueError("Sample larger than population or is negative")
    322         result = [None] * k
    323         setsize = 21        # size of a small set minus size of an empty list

ValueError: Sample larger than population or is negative
```

So, in all its essence we need to put a validation before sample() is called.

**Comment by makcedward** [comment]
Fixed in BETA release. You may try again. It will be included 0.0.4 release.

**Comment by neerajvashistha** [comment]
thanks :)
